sunpy/lightcurve/sources/soho.py

```diff
@@ -0,0 +1,158 @@
+        #     ax2.set_yscale("log")
+        #     ax2.set_ylim(1e-9, 1e-2)
+        #     ax2.set_yticks((1e-9, 1e-8, 1e-7, 1e-6, 1e-5, 1e-4, 1e-3, 1e-2))
+        #     ax2.set_yticklabels((' ', 'A', 'B', 'C', 'M', 'X', ' '))
+        
+        #     axes.yaxis.grid(True, 'major')
+        #     axes.xaxis.grid(False, 'major')
+        #     axes.legend()
+        
+        #     # @todo: display better tick labels for date range (e.g. 06/01 - 06/05)
+        #     formatter = matplotlib.dates.DateFormatter('%H:%M')
+        #     axes.xaxis.set_major_formatter(formatter)
+        
+        #     axes.fmt_xdata = matplotlib.dates.DateFormatter('%H:%M')
+        #     figure.autofmt_xdate()
+        #     figure.show()
+        
+        #     return figure
+
+
+        @staticmethod
+        def _parse_txt(filepath):
+                """
+                Parses a SOHO/ERNE file from
+                http://srl.utu.fi/erne_data/carrot/carrota.html
+                http://srl.utu.fi/erne_data/carrot/carrotp.html
+                
+                and returns header as a list and ERNE data as pandas dataframe
+                """
+                
+                #Reading in Data along with header
+                data = ascii.read(filepath, delimiter = "\s", data_start = 2 ) 
+                
+                #header is read along with data i.e the first row in data is the header
+                header = data.colnames
+                #So extracting the first row elements and forming header list that contains the energy bins
+                header = [data[0][key] for key in data.colnames]
+                #and now excluding the first line to have only the data in data variable
+                data = data[1:]
+                
+                data_modify = []
+                
+                #Storing data columns in recognizable variables
+                year_col = data['col1']
+                month_col = data['col2']
+                date_col = data['col3']
+                begin_time_col = data['col4']
+                end_time_col = data['col5']
+                
+                #Converting separate datetime element into a single datetime.datetime column
+                for i in range(len(data)):
+                    
+                        #Combining separate datetime elements into single datetime value
+                        #start time
+                        date1 = datetime.strptime(year_col[i] + '-' + month_col[i] + '-' + date_col[i] + '/' + begin_time_col[i][:2] + ':' +begin_time_col[i][2:],"%y-%m-%d/%H:%M")
+                        #end time
+                        date2 = datetime.strptime(year_col[i] + '-' + month_col[i] + '-' + date_col[i] + '/' + end_time_col[i][:2] + ':' + end_time_col[i][2:], "%y-%m-%d/%H:%M")
+                        #Appending the start and end time as sunpy.time TimeRange in a separate list
+                        data_modify.append(TimeRange(date1, date2))
+                
```

Comment:
oops....got it !!!
ᐧ

On Thu, Jul 2, 2015 at 3:24 PM, dpshelio <notifications@github.com> wrote:

> In sunpy/lightcurve/sources/soho.py
> <https://github.com/sunpy/sunpy/pull/1476#discussion_r33762883>:
>
> > +                month_col = data['col2']
> > +                date_col = data['col3']
> > +                begin_time_col = data['col4']
> > +                end_time_col = data['col5']
> > +
> > +                #Converting separate datetime element into a single datetime.datetime column
> > +                for i in range(len(data)):
> > +
> > +                        #Combining separate datetime elements into single datetime value
> > +                        #start time
> > +                        date1 = datetime.strptime(year_col[i] + '-' + month_col[i] + '-' + date_col[i] + '/' + begin_time_col[i][:2] + ':' +begin_time_col[i][2:],"%y-%m-%d/%H:%M")
> > +                        #end time
> > +                        date2 = datetime.strptime(year_col[i] + '-' + month_col[i] + '-' + date_col[i] + '/' + end_time_col[i][:2] + ':' + end_time_col[i][2:], "%y-%m-%d/%H:%M")
> > +                        #Appending the start and end time as sunpy.time TimeRange in a separate list
> > +                        data_modify.append(TimeRange(date1, date2))
> > +
>
> Also... Is it there just a file per day? what happens if one measurement
> ends in the following day? say, the end time is 0003 as it's above the
> Timerange would be negative...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sunpy/sunpy/pull/1476/files#r33762883>.
>